keyboards/ferris/keymaps/venc0r/keymap.c

```diff
@@ -0,0 +1,100 @@
+#include "venc0r.h"
+
+#define LAYOUT_wrapper(...) LAYOUT(__VA_ARGS__)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [_ALPHA] = LAYOUT_wrapper(
+        _______________S_COLEMAK_L0________________, _______________S_COLEMAK_R0________________,
+        _______________S_COLEMAK_L1________________, _______________S_COLEMAK_R1________________,
+        _______________S_COLEMAK_L2________________, _______________S_COLEMAK_R2________________,
+        _______________S_COLEMAK_L3________________, _______________S_COLEMAK_R3________________,
+        __S_THUMB_L____, __S_THUMB_R____
+    ),
+
+    [_LOWER] = LAYOUT_wrapper(
+        _______________S_SYMBOL_L0_________________, _______________S_SYMBOL_R0_________________,
+        _______________S_SYMBOL_L1_________________, _______________S_SYMBOL_R1_________________,
+        _______________S_SYMBOL_L2_________________, _______________S_SYMBOL_R2_________________,
+        _______________S_SYMBOL_L3_________________, _______________S_SYMBOL_R3_________________,
+        __S_THUMB_N____, __S_THUMB_N____
+    ),
+
+    [_RAISE] = LAYOUT_wrapper(
+        _______________S_NAV_L0____________________, _______________S_NAV_R0____________________,
+        _______________S_NAV_L1____________________, _______________S_NAV_R1____________________,
+        _______________S_NAV_L2____________________, _______________S_NAV_R2____________________,
+        _______________S_NAV_L3____________________, _______________S_NAV_R3____________________,
+        __S_THUMB_N____, __S_THUMB_N____
+    ),
+
+    [_ADJUST] = LAYOUT_wrapper(
+        _______________S_ADJUST_L0_________________, _______________S_ADJUST_R0_________________,
+        _______________S_ADJUST_L1_________________, _______________S_ADJUST_R1_________________,
+        _______________S_ADJUST_L2_________________, _______________S_ADJUST_R2_________________,
+        _______________S_ADJUST_L3_________________, _______________S_ADJUST_R3_________________,
+        __S_THUMB_N____, __S_THUMB_N____
+    ),
+
+    [_GAME] = LAYOUT_wrapper(
+        _______________S_GAME_L0___________________, _______________S_GAME_R0___________________,
+        _______________S_GAME_L1___________________, _______________S_GAME_R1___________________,
+        _______________S_GAME_L2___________________, _______________S_GAME_R2___________________,
+        _______________S_GAME_L3___________________, _______________S_GAME_R3___________________,
+        __S_G_THUMB_L__, __S_G_THUMB_R__
+    ),
+};
+
+#ifdef ENCODER_ENABLE
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
```

Comment:
You probably want to use encoder maps here, as that's doing basically what you have specified. 

https://docs.qmk.fm/#/feature_encoders?id=encoder-map